Lib/test/test_posix.py

```diff
@@ -1476,6 +1478,106 @@ def test_empty_file_actions(self):
         )
         self.assertEqual(os.waitpid(pid, 0), (pid, 0))
 
+    def test_resetids(self):
+        pid = posix.posix_spawn(
+            sys.executable,
+            [sys.executable, '-c', 'pass'],
+            os.environ,
+            resetids=True
+        )
+
+        self.assertEqual(os.waitpid(pid, 0), (pid, 0))
+
+
+    def test_resetids_wrong_type(self):
+        with self.assertRaises(TypeError):
+            posix.posix_spawn(sys.executable,
+                              [sys.executable, "-c", "pass"],
+                              os.environ, resetids=0)
+        with self.assertRaises(TypeError):
+            posix.posix_spawn(sys.executable,
+                              [sys.executable, "-c", "pass"],
+                              os.environ, resetids=1.0)
+        with self.assertRaises(TypeError):
+            posix.posix_spawn(sys.executable,
+                              [sys.executable, "-c", "pass"],
+                              os.environ, resetids="True")
+
+    def test_setpgroup(self):
+        pid = posix.posix_spawn(
+            sys.executable,
+            [sys.executable, '-c', 'pass'],
+            os.environ,
+            setpgroup=os.getpgrp()
+        )
+
+        self.assertEqual(os.waitpid(pid, 0), (pid, 0))
+
+
+    def test_setsigmask_wrong_type(self):
+        with self.assertRaises(TypeError):
+            posix.posix_spawn(sys.executable,
+                              [sys.executable, "-c", "pass"],
+                              os.environ, setsigmask=["j","k","k"])
+        with self.assertRaises(TypeError):
+            posix.posix_spawn(sys.executable,
+                              [sys.executable, "-c", "pass"],
+                              os.environ, setsigmask=[3+4j])
+        with self.assertRaises(TypeError):
+            posix.posix_spawn(sys.executable,
+                              [sys.executable, "-c", "pass"],
+                              os.environ, setsigmask="iuhuhiuh")
+
+    def test_setsigmask(self):
```

Comment:
I had a more complete test for this function that worked on my machine but not on Travis. I suspect is because how the signal handlers are inherited but I am not sure. The test was:

```python
     def test_setsigmask(self):
         code = textwrap.dedent("""\
              import sys
              def handler(signum, frame):
                  sys.exit(-1)

              import signal
              signal.signal(signal.SIGUSR1, handler)
              import time
              time.sleep(0.1)""")

         pid = posix.posix_spawn(
             sys.executable,
             [sys.executable, '-c', code],
             os.environ,
             setsigmask=[signal.SIGUSR1]
         )

         os.kill(pid, signal.SIGUSR1)
         self.assertEqual(os.waitpid(pid, 0), (pid, 0))
```